bundles/org.openhab.binding.tacmi/src/main/java/org/openhab/binding/tacmi/internal/schema/TACmiSchemaHandler.java

```diff
@@ -0,0 +1,295 @@
+                        + (command == OnOffType.ON ? "1" : "0")));
+                reqUpdate.header(HttpHeader.REFERER, this.serverBase + "schema.html"); // required...
+                break;
+            case SWITCH_FORM:
+                ChangerX2Entry cx2e = e.changerX2Entry;
+                if (cx2e != null) {
+                    reqUpdate = prepareRequest(buildUri("INCLUDE/change.cgi?changeadrx2=" + cx2e.address
+                            + "&changetox2=" + (command == OnOffType.ON ? "1" : "0")));
+                    reqUpdate.header(HttpHeader.REFERER, this.serverBase + "schema.html"); // required...
+                } else {
+                    logger.warn("Got command for uninitalized channel {}: {}", channelUID, command);
+                    return;
+                }
+                break;
+            case STATE_FORM:
+                ChangerX2Entry cx2sf = e.changerX2Entry;
+                if (cx2sf != null) {
+                    String val = cx2sf.options.get(((StringType) command).toFullString());
+                    if (val != null) {
+                        reqUpdate = prepareRequest(
+                                buildUri("INCLUDE/change.cgi?changeadrx2=" + cx2sf.address + "&changetox2=" + val));
+                        reqUpdate.header(HttpHeader.REFERER, this.serverBase + "schema.html"); // required...
+                    } else {
+                        logger.warn("Got unknown form command {} for channel {}; Valid commands are: {}", command,
+                                channelUID, cx2sf.options.keySet());
+                        return;
+                    }
+                } else {
+                    logger.warn("Got command for uninitalized channel {}: {}", channelUID, command);
+                    return;
+                }
+                break;
+            case READ_ONLY_NUMERIC:
+            case READ_ONLY_STATE:
+            case READ_ONLY_SWITCH:
+                logger.warn("Got command for ReadOnly channel {}: {}", channelUID, command);
+                return;
+            default:
+                logger.warn("Got command for unhandled type {} channel {}: {}", e.type, channelUID, command);
+                return;
+        }
+        try {
+            ContentResponse res = reqUpdate.send();
+            if (res.getStatus() == 200) {
+                // update ok, we update the state
+                e.setLastState((State) command);
+                updateState(channelUID, (State) command);
+            } else {
+                logger.error("Error sending update for {} = {}: {} {}", channelUID, command, res.getStatus(),
+                        res.getReason());
+            }
+        } catch (InterruptedException | TimeoutException | ExecutionException ex) {
+            logger.error("Error sending update for {} = {}: {}", channelUID, command, ex.getMessage(), ex);
```

Comment:
Protocol errors shouldn't be logged to error. This could be warn. Also it seems unreasonable to log the stack trace when the network fails.